scrapper.py

```diff
@@ -21,21 +34,57 @@ class IncorrectNumberOfArticlesError(Exception):
     """
 
 
+class ResponseIsNotOk(Exception):
+    """
+    Response status code is not ok
+    """
+
+
 class Crawler:
     """
     Crawler implementation
     """
-    def __init__(self, seed_urls, max_articles: int):
-        pass
+
+    def __init__(self, seed_urls, total_max_articles: int):
+        self.seed_urls = seed_urls
+        self.total_max_articles = total_max_articles
+        self.urls = []
+
+    @staticmethod
+    def _changing_the_link(link):
+        if 'http://journals.tsu.ru' not in link:
+            link = 'http://journals.tsu.ru' + link
+        return link
 
     def _extract_url(self, article_bs):
-        pass
+        urls_bs = article_bs.find('div', class_='two_thirds')
+        urls_bs = urls_bs.find_all('a')
+        urls_bs_full = [self._changing_the_link(url_bs['href']) for url_bs in urls_bs]
+        return urls_bs_full
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        for seed_url_index, seed_url in enumerate(self.seed_urls):
+            response = requests.get(url=seed_url, headers=HEADERS)
+
+            if not response.ok:
+                raise ResponseIsNotOk
```

Comment:
and who will handle it? you should be more tolerant. Your goal is to collect enough quantity of valid article URLs, you should be targeting this goal, rather than failing at the first non-working link